Рабочая игра. Ментор Дыдыкин

maratkanov-a opened the pull request.

inerc committed:
Create README.md

inerc committed:
Update README.md

smirnov-vs committed:
paths

inerc committed:
SW work oll games

inerc committed:
Merge branch 'master' of https://github.com/inerc/2016_1_ShockWork

inerc committed:
ReqJS and Grunt

inerc committed:
Merge branch 'master' of https://github.com/inerc/2016_1_ShockWork

inerc committed:
Update README.md

inerc committed:
Update README.md

inerc committed:
add less for template.css

inerc committed:
Merge branch 'master' of https://github.com/inerc/2016_1_ShockWork

inerc committed:
Update README.md

inerc committed:
Update README.md

inerc committed:
add MediaBackground

inerc committed:
Merge branch 'master' of https://github.com/inerc/2016_1_ShockWork

smirnov-vs committed:
game

smirnov-vs committed:
img

maratkanov-a committed:
fixed registration and login

smirnov-vs committed:
fix

maratkanov-a committed:
added main loader

maratkanov-a committed:
fixed main loader

maratkanov-a committed:
fixed loaders

smirnov-vs committed:
ssl

maratkanov-a committed:
index refactor

smirnov-vs committed:
.

smirnov-vs committed:
...

maratkanov-a committed:
Merge branch 'master' of https://github.com/shramru/2016_1_ShockWork into shramru-master

maratkanov-a committed:
Merge branch 'shramru-master'

smirnov-vs committed:
fix

maratkanov-a committed:
1

maratkanov-a committed:
1

maratkanov-a committed:
1

maratkanov-a committed:
1

maratkanov-a committed:
1

maratkanov-a committed:
1

madimp commented on public_html/js/views/offline.js:
```diff
@@ -303,16 +532,17 @@ define([
-            if (stack.length < 3) count = stack.length;
+            cards_counter = 0;
+            count = 3;
+            if (stack.length < 3) var count = stack.length;
```
наказать и распять

madimp commented on public_html/js/views/offline.js:
```diff
@@ -303,16 +532,17 @@ define([
             var newThis = this.$el;
             for (var i=0; i < count; i++ ){
-                $('<li class="ui-state-default"><img src="img/cards/'+stack[i].img+'.png" alt=""> </li>')
+                $('<li class="ui-state-default"  style = "list-style: none;"><img src="img/cards/'+stack[i].img+'.png" alt=""> </li>')
```
верстка в скрипте это еще один сеанс насаживания на кол

madimp commented on public_html/js/views/registration.js:
```diff
@@ -59,48 +59,48 @@ define([
-        	$(resnap).toggle();
+            $(canvas).toggle();
+            $(video).toggle();
+            $(snap).toggle();
```
ужас!))

madimp commented on templates/game.xml:
```diff
@@ -1,6 +1,7 @@
 <fest:template xmlns:fest="http://fest.mail.ru" context_name="json">
 
     <link rel="stylesheet" href="css/cards/jquery-ui.css" />
+    <link rel="stylesheet" href="css/cards/reset.css" />
```
все поконкатить

inerc committed:
fix

inerc committed:
fix

smirnov-vs committed:
Merge remote-tracking branch 'upstream/master'

inerc committed:
fix problem

inerc committed:
fix problem

maratkanov-a committed:
Merge pull request #14 from inerc/master

Рабочий проект

smirnov-vs committed:
Merge remote-tracking branch 'upstream/master'

smirnov-vs committed:
fixed define

smirnov-vs committed:
some fixes

inerc committed:
Update main.css

inerc committed:
Merge branch 'master' of https://github.com/maratkanov-a/2016_1_ShockWork

inerc committed:
Распределил жизни по двум сторонам

maratkanov-a committed:
Merge pull request #15 from shramru/master

фиксы

sp41mer committed:
made normal flame and refactored some code

inerc committed:
Add User VS User

smirnov-vs committed:
preloader

smirnov-vs committed:
fix

maratkanov-a committed:
Merge pull request #17 from shramru/master

preloader

sp41mer committed:
made something (uzhe ne pomnu)

smirnov-vs committed:
music destroy. sorry, Jenek

sp41mer committed:
Merge remote-tracking branch 'upstream/master'

maratkanov-a committed:
Merge pull request #18 from shramru/master

music destroy

sp41mer committed:
Merge remote-tracking branch 'upstream/master'

maratkanov-a committed:
Merge pull request #19 from sp41mer/master

Поправил игру

maratkanov-a committed:
superhotfix

inerc committed:
Merge Maratcanov

maratkanov-a committed:
fixed scoreboard

smirnov-vs committed:
поправил ужас

maratkanov-a committed:
fixed canvas

smirnov-vs committed:
minify

smirnov-vs committed:
v2

smirnov-vs committed:
Merge remote-tracking branch 'upstream/master'

maratkanov-a committed:
Merge pull request #20 from shramru/master

фичи

maratkanov-a commented:
UPDATE

madimp commented on public_html/appCache.js:
```diff
@@ -1,10 +1,59 @@
+                  '/',
+                  '/index.html',
+                  '/js/build.min.js',
+				  '/js/app.js',
```
может отступы?

madimp commented on public_html/config.js:
```diff
@@ -0,0 +1,27 @@
+    },
+    shim: {
+        'backbone': {
+            deps: ['underscore', 'jquery', 'jquery_ui'],
```
уберите отсюда jquery_ui
бэкбон от него не зависит

madimp commented on public_html/js/main.js:
```diff
@@ -1,13 +1,13 @@
+        sweetalert: 'lib/sweetalert.min'
     },
     shim: {
         'backbone': {
```
jquery_ui убрать
в чем смысл main.js config.js и app.js?

madimp commented on public_html/js/views/game.js:
```diff
@@ -18,61 +15,212 @@ define([
 
         initialize: function () {
+            this.showed = false;
             this.render();
```
зачем рендерить игру при заходе в главное меню?

smirnov-vs replied:
а когда лучше? поставить флаг на первый рендер при заходе?

madimp replied:
игру когда рендерить? когда пошел играть

madimp commented on public_html/js/views/game.js:
```diff
@@ -18,61 +15,212 @@ define([
             this.draw(this.user1_stack);
-            this.draw_enemy(this.user2_stack_length)
+            this.draw_enemy(this.user2_stack_length);
+            $('#button_done').show();
```
какую кнопку ищем? любую, где бы она ни была?

madimp commented on public_html/js/views/game.js:
```diff
@@ -18,61 +15,212 @@ define([
+            this.init_table();
+            this.$el.find('#button_done').show();
+            this.$el.find('#restart_button').hide();
+            $(".not_my").text('?');
```
что ищем и где?

madimp commented on public_html/js/views/game.js:
```diff
@@ -18,61 +15,212 @@ define([
+            }
+        },
+        showHealth: function(msgData){
+            $(".not_my").text(msgData.enemyPower);
```
аналогично

madimp commented on public_html/js/views/game.js:
```diff
@@ -18,61 +15,212 @@ define([
+            $(".my").text(msgData.power);
+        },
+        showPower: function(msgData){
+            $(".not_my").text(msgData.enemyPower);
```
аналогично

madimp commented on public_html/js/views/game.js:
```diff
@@ -18,61 +15,212 @@ define([
         },
         init_table: function() {
-            $(".score span").text('0');
+            $(".score span.my").text('0');
```
аналогично

madimp commented on public_html/js/views/game.js:
```diff
@@ -18,61 +15,212 @@ define([
             var newThis = this.$el;
             for (var i = 1; i <= 3; i++) {
-                $('<div style = "height: 205px; width: 100%"> </div>')
+                $('<div style = "height: 180px; width: 100%"> </div>')
```
убрать верстку из js

madimp commented on public_html/js/views/game.js:
```diff
@@ -93,14 +241,14 @@ define([
             ui.draggable.draggable( 'option', 'revert', false );
-            $(".score span").text(parseInt($(".score span").text())+cardPower);
             ui.draggable.data('this').USER_power += cardPower;
+            $(".score span.my").text(ui.draggable.data('this').USER_power);
```
где?

madimp commented on public_html/js/views/game.js:
```diff
@@ -93,14 +241,14 @@ define([
             var newThis = this.$el;
             for (var i=0; i < count; i++ ){
-                $('<li class="ui-state-default"><img src="img/cards/'+stack[i].img+'.png" alt=""> </li>')
+                $('<li class="ui-state-default" style = "list-style: none;"><img src="img/cards/'+stack[i].img+'.png" alt=""> </li>')
```
убрать верстку из js

madimp commented on public_html/js/views/login.js:
```diff
@@ -60,7 +58,7 @@ define([
                 this.$el.find('.form__error').hide();
                 valid.forEach(function (item) {
-                    $('.form__'+ item +'__error').text("Required").show()
+                    $('.form__'+ item +'__error').text("Обязательное поле").show()
```
где?

madimp commented on public_html/js/views/main.js:
```diff
@@ -17,6 +16,7 @@ define([
         show: function() {
             this.$el.show();
             this.trigger("show",this);
+            $('body').addClass('loaded');
```
и игра и главное меню вешают и убирают этот класс. в чем понт?

maratkanov-a replied:
это загрузчик такой

madimp commented on public_html/js/views/offline.js:
```diff
@@ -278,10 +293,10 @@ define([
             var newThis = this.$el;
             for (var i = 1; i <= 3; i++) {
-                $('<div style = "height: 205px; width: 100%"> </div>').
+                $('<div style = "height: 180px; width: 100%"> </div>').
```
убрать верстку из js

madimp commented on public_html/js/views/offline.js:
```diff
@@ -303,16 +318,19 @@ define([
             var newThis = this.$el;
             for (var i=0; i < count; i++ ){
-                $('<li class="ui-state-default"><img src="img/cards/'+stack[i].img+'.png" alt=""> </li>')
+                $('<li class="ui-state-default"  style = "list-style: none;"><img src="img/cards/'+stack[i].img+'.png" alt=""> </li>')
```
убрать

madimp commented on public_html/js/views/offline.js:
```diff
@@ -347,28 +366,35 @@ define([
+
             }
             if (this.USER_health <= 0) {
                 alert('you loose');
```
правильно, что вы не идете на защиту )))

inerc replied:
Мы все сделаем и пойдем)

smirnov-vs replied:
Я всю ночь сидел и правил. Посидим еще парочку и пойдем!

madimp replied:
ну тогда, чтоб ни одного алерта в коде! ))

madimp commented on public_html/js/views/scoreboard.js:
```diff
@@ -28,6 +26,10 @@ define([
             this.trigger("show",this);
+            ScoresCollection.fetch({remove: false})
+                .done( function(){
+                    newThis.$el.html(newThis.template( { scores : ScoresCollection.toJSON() } ));
```
вот эта хрень это рендер
а в рендере быть запроса данных не должно быть

smirnov-vs replied:
А как можно обновить коллекцию очков без рендера? Иначе всегда одни результаты

madimp replied:
на показе фетч, на саксесе рендер.
а не и в рендере и в показе фетч с перерисовкой на саксесе
рендер делает только рендер того, что есть в коллекции
обновить ее не его задача

madimp commented on templates/game.xml:
```diff
@@ -1,78 +1,78 @@
+   </div>
+   <!--==============================================-->
+   <div id = "waiter"
+      style = "width:100%;
```
стили в стили

madimp commented on templates/game.xml:
```diff
@@ -1,78 +1,78 @@
+                     <ul id="sortable2" class="connectedSortable">
+                     </ul>
+                  </div>
+                  <div class = "score" style = "width: 120px; height:120px; color: #940d13; position: absolute;
```
аналогично

madimp commented on templates/game.xml:
```diff
@@ -1,78 +1,78 @@
+                     <span>HP:</span><span id = "your_health">50</span>
+                     <span>Enemy:</span><span id = "enemy_health">50</span>
+                  </div>
+                  <div id = "restart_button" align = "center" style = "position: absolute;
```
ага

sp41mer committed:
cho-to fixil

smirnov-vs committed:
fix minify

smirnov-vs committed:
fix

sp41mer committed:
Sdelal pravki Didikina

Merge remote-tracking branch 'upstream/master'

# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

smirnov-vs committed:
fix js files

smirnov-vs committed:
fix css

maratkanov-a committed:
Merge pull request #21 from shramru/master

новые фиксы

smirnov-vs committed:
fix url material

smirnov-vs committed:
remove

sp41mer committed:
Sdelal smenu knopki i che-to fixil

Merge remote-tracking branch 'upstream/master'

# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

smirnov-vs committed:
отступы

maratkanov-a committed:
Merge pull request #22 from shramru/master

еще порция

sp41mer committed:
Merge remote-tracking branch 'upstream/master'

sp41mer committed:
Merge remote-tracking branch 'upstream/master'

sp41mer committed:
button

smirnov-vs committed:
font

madimp commented on public_html/js/config.js:
```diff
@@ -23,5 +22,4 @@ require.config({
             deps: ['jquery', 'hammerjs']
```
underscore не зависит от hammerjs :)

smirnov-vs replied:
этож materialize)

madimp replied:
сорян ))

sp41mer committed:
fixed missed css

maratkanov-a committed:
Merge pull request #23 from sp41mer/master

Надеюсь сейчас я не обосрался

smirnov-vs committed:
Merge remote-tracking branch 'upstream/master'

sp41mer committed:
put swals instead of alerts

sp41mer committed:
made some verstka

smirnov-vs committed:
Merge pull request #24 from sp41mer/master

Надеюсь не обосрался (п.2)

smirnov-vs committed:
Merge remote-tracking branch 'upstream/master'

sp41mer committed:
button done now hidden and fixed gray layer

sp41mer committed:
Merge remote-tracking branch 'upstream/master'

sp41mer committed:
sdelal all verstka krome loader

sp41mer committed:
fixed sockets

sp41mer committed:
fixed sockets №2

maratkanov-a committed:
Merge pull request #27 from sp41mer/master

Вот теперь точно подъехала

smirnov-vs committed:
Merge remote-tracking branch 'upstream/master'

sp41mer committed:
Fixed button

Sdelal smenu knopki i che-to fixil

Merge remote-tracking branch 'upstream/master'

# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

smirnov-vs committed:
Merge pull request #28 from sp41mer/master

Fixed button

smirnov-vs committed:
Merge remote-tracking branch 'upstream/master'

smirnov-vs committed:
fix done button

smirnov-vs committed:
Merge pull request #29 from shramru/master

kek

smirnov-vs committed:
еще фиксы

smirnov-vs committed:
Merge pull request #30 from shramru/master

еще фиксы

smirnov-vs committed:
fix offline

smirnov-vs committed:
fixxx

smirnov-vs committed:
hotfix

smirnov-vs committed:
superhotfix

smirnov-vs committed:
Merge pull request #31 from shramru/master

pullllllllllll

smirnov-vs committed:
cache

madimp commented on public_html/css/build.css:
```diff
@@ -1363,20 +1363,113 @@ body.stop-scrolling {
 .nav-top {
   margin-top: 200px;
 }
+.js-insert-back{
```
классы с префиксом js для поиска элементов из скриптов. на них не должно быть стилей

madimp commented on public_html/js/views/game.js:
```diff
@@ -198,7 +187,7 @@ define([
             var newThis = this.$el;
             for (var i=0; i < msgData.enemyCards.length; i++ ){
-                $('<li class="ui-state-default" style = "list-style: none;"><img src="img/cards/'+msgData.enemyCards[i].img+'.png" alt=""> </li>')
+                $('<li class="ui-state-default"><img src="img/cards/'+msgData.enemyCards[i].img+'.png" alt=""> </li>')
```
верстку в шаблоны

madimp commented on public_html/js/views/game.js:
```diff
@@ -243,13 +233,14 @@ define([
             var newThis = this.$el;
             for (var i=0; i < count; i++ ){
-                $('<li class="ui-state-default" style = "list-style: none;"><img src="img/cards/'+stack[i].img+'.png" alt=""> </li>')
+                $('<li class="ui-state-default"><img src="img/cards/'+stack[i].img+'.png" alt=""> </li>')
```
верстку в шаблоны

maratkanov-a committed:
not working fix

smirnov-vs committed:
fix cards

inerc committed:
fix problem Kozinchenko

inerc committed:
Fix main css

inerc committed:
merge inerc

smirnov-vs committed:
fix

inerc committed:
Merge remote-tracking branch 'origin/master'

inerc committed:
fix problem Inerc

smirnov-vs committed:
fix render

smirnov-vs committed:
fix registartion

inerc committed:
fix js inerc

inerc committed:
merge inerc/maratckanov

smirnov-vs committed:
Merge pull request #32 from inerc/master

кук пусики

smirnov-vs committed:
Revert "кук пусики"

smirnov-vs committed:
Merge pull request #33 from maratkanov-a/revert-32-master

Revert "кук пусики"

inerc committed:
fix main.css

inerc committed:
удалили бред из css

inerc committed:
Вроде все четко;)

inerc committed:
fix problem js

smirnov-vs committed:
Merge pull request #34 from inerc/master

Поправил, чекни

smirnov-vs committed:
fix backs

smirnov-vs committed:
fix naming

smirnov-vs committed:
move

smirnov-vs commented:
Merge request

smirnov-vs committed:
new cards

maratkanov-a commented:
UPDATE

smirnov-vs committed:
game improvements

smirnov-vs committed:
hotfix

smirnov-vs committed:
mana win
